dhcpv4/options.go

```diff
@@ -191,6 +191,12 @@ func (o Options) Marshal(b *uio.Lexer) {
 
 		data := o[code]
 
+		// Ensure even 0-length options are written out
+		if len(data) == 0 {
+			b.Write8(code)
+			b.Write8(0)
```

Comment:
0 length options like Pad do not get a length field.

Reply:
What is option 224?

Reply:
RFC 2132 has:
> Any options defined subsequent to this document MUST contain a length octet even if the length is fixed or zero.

Padding is a special case (also with End, the only 0-length option defined *before* RFC2132), I don't remember if it is handled differently in the code. I'll check

For option 224, it's in the private use area, just used here for the test. Another possibility would be option 80 rapid commit, which *does* have a length field per [RFC4039§4](https://tools.ietf.org/html/rfc4039#section-4)

Reply:
Handled Pad similarly to End, by ignoring it; since we don't store the order of options and sort them by code anyway, padding doesn't make sense as it could be moved around anywhere.

Reply:
ok makes sense. thanks